File: src/main/java/com/dude/dms/backend/repositories/TagHistoryRepository.java
package com.dude.dms.backend.repositories;

import com.dude.dms.backend.data.Tag;
import com.dude.dms.backend.data.history.TagHistory;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface TagHistoryRepository extends JpaRepository<TagHistory, Long> {

    List<TagHistory> findByTag(Tag tag);
}

File: src/main/java/com/dude/dms/backend/repositories/TagHistoryRepository.kt
package com.dude.dms.backend.repositories

import com.dude.dms.backend.data.Tag
import com.dude.dms.backend.data.history.TagHistory
import org.springframework.data.jpa.repository.JpaRepository

interface TagHistoryRepository : JpaRepository<TagHistory, Long> {

    fun findByTag(tag: Tag): Set<TagHistory>
}